### ISSUE TITLE
-i[ext] like perl

### ISSUE THREAD

**Issue by Freed-Wu** [wader/fq#1019]
I want to use fq to replace perl, such as:

```sh
perl -pes=a=b= test.txt
fq -Rr 'sub("a"; "b")' test.txt
```

However, perl have `-i`

```sh
perl -i.bak -pes=a=b= test.txt
ls test.txt.bak
```

How can fq support it?

**Comment by wader** [comment]
Hi, i'm a bit reluctant to add this before jq adds it as i would like to do it in the same way. For jq there has been an attempt to do it that got reverted and there has been some quite lively discussion about it, see https://github.com/jqlang/jq/issues?q=inplace

As i understand it there are some issues and behaviours to think about:
- Should it work on just one file (like https://github.com/mikefarah/yq)? if not, how to handle errors? revert previous written files? can't be done atomically?
- Output JSON will be reformatted, keys sorted and comments will be stripped
- fq currently uses `-i` for interactive REPL 🤔 maybe use `-I`or `--in-place`?
- Only for JSON input? not sure it would make sense to do binary in JSON out?

If i would do this i think a first implementation would be first file only (and JSON value not spanning multiple files) and only JSON in/out. 

Any ideas how it could work?

**Comment by Freed-Wu** [comment]
> reluctant to add this before jq adds it as i would like to do it in the same way

So we should wait https://github.com/jqlang/jq/issues/105 :smile:

**Comment by wader** [comment]
Yeah i guess so, sadly it might take some time 😬 but who knows! but would it be ok in your use case if the json get reformatted? (i mentioned comments, but i confused it with reformatting jq code 😄 )
